easybuild/tools/github.py

```diff
@@ -1073,17 +1073,35 @@ def find_software_name_for_patch(patch_name, ec_dirs):
     for ec_dir in ec_dirs:
         for (dirpath, _, filenames) in os.walk(ec_dir):
             for fn in filenames:
-                if fn != 'TEMPLATE.eb' and not fn.endswith('.py'):
+                # TODO: In EasyBuild 5.x only check for '*.eb' files
+                if fn != 'TEMPLATE.eb' and os.path.splitext(fn) not in ('.py', '.patch'):
```

Comment:
This returns a tuple
```suggestion
                if fn != 'TEMPLATE.eb' and os.path.splitext(fn)[1] not in ('.py', '.patch'):
```
I'm testing out this code with a PR where I'm currently getting stuck.

Reply:
eh, actually, not even this is enough. 
I had git repo in path, and it's searching through `.git/index` and failing to parse it as utf-8. This should really have the same logic is robot does

Reply:
I printed the paths just to see what it is trying to read, and it's a lot. E.g.
```
Reading /local/EB/build/eb-hqbrnpbu/git-working-dird1p5bheb/easybuild-easyconfigs/easybuild/easyconfigs/0/3to2/3to2-1.1.1-foss-2016b-Python-2.7.12.eb
Reading /local/EB/build/eb-hqbrnpbu/git-working-dird1p5bheb/easybuild-easyconfigs/.github/workflows/unit_tests.yml
Reading /local/EB/build/eb-hqbrnpbu/git-working-dird1p5bheb/easybuild-easyconfigs/.github/workflows/linting.yml
Reading /local/EB/build/eb-hqbrnpbu/git-working-dird1p5bheb/easybuild-easyconfigs/.git/index
```

Reply:
To be clear, I'm printing all the paths that "slip through" the extension filtering here 
```python
                # TODO: In EasyBuild 5.x only check for '*.eb' files
                if fn != 'TEMPLATE.eb' and os.path.splitext(fn)[1] '.eb')[1] not in ('.py', '.patch'):
                    path = os.path.join(dirpath, fn)
                    print('Reading', path)
                    rawtxt = read_file(path)
                    if 'patches' in rawtxt:
                        all_ecs.append(path)
```

Reply:
So, `ec_dirs = ['/local/EB/build/eb-7c1srdq1/git-working-dirgr_pgwkk/easybuild-easyconfigs']` is the problem then. It's 2 directories to high up.

Reply:
I suppose we can add the 2 folders to `patch_specs = det_patch_specs(paths['patch_files'], file_info, [target_dir])` but for now, to keep it generic, I exclude hidden folders which also works

Reply:
Using the ignore-dirs option now which includes .git by default. :)